### PR TITLE
Avoid overwriting runpath keyword during loop

### LINKED ISSUE #4569: SPE1 fails due to RUNPATH overwritten
test spe1 failed this night with the following error

```
 Traceback (most recent call last):
  File "/prog/res/komodo/bleeding-py38-rhel7/root/lib64/python3.8/site-packages/ert/shared/share/ert/forward-models/res/script/ecl100.py", line 9, in <module>
    run(config, [arg for arg in sys.argv[1:] if len(arg) > 0])
  File "/prog/res/komodo/bleeding-py38-rhel7/root/lib64/python3.8/site-packages/ert/shared/share/ert/forward-models/res/script/ecl_run.py", line 484, in run
    run = EclRun(
  File "/prog/res/komodo/bleeding-py38-rhel7/root/lib64/python3.8/site-packages/ert/shared/share/ert/forward-models/res/script/ecl_run.py", line 208, in __init__
    raise IOError(f"No such file: {data_file}")
OSError: No such file: eclipse/SPE1CASE1.DATA
```

Indeed, SPE1CASE1.DATA is not present in the real-0/iter-0 directory. Looking into the copy_directory.stdout.0 file says

Copying directory structure /tmp/jenkins-komodo-f_scout_ci/workspace/x-komodo-test-spe1/opm-data/spe1 -> /tmp/jenkins-komodo-f_scout_ci/workspace/x-komodo-test-spe1/spe1_out/real_1/iter_0/eclipse

which shows it tries to copy to the real_1 instead. [config-file](https://github.com/equinor/komodo-releases/blob/main/jenkins/spe1/spe1.ert) says

```
QUEUE_SYSTEM LOCAL
QUEUE_OPTION LOCAL MAX_RUNNING 50

RUNPATH spe1_out/real_%d/iter_%d

NUM_REALIZATIONS 2

ECLBASE eclipse/SPE1CASE1

FORWARD_MODEL COPY_DIRECTORY(<FROM>=<CONFIG_PATH>/opm-data/spe1, <TO>=<RUNPATH>/eclipse)

FORWARD_MODEL ECLIPSE100(<VERSION>=2017.1)
```

so i expect this to be related to the change from %d . Could you
[@andrli](https://equinor.slack.com/team/U043VPLFU8K)
verify the behavior is as intended here, and eventually update the config-file to reflect how it should be?
[spe1.ert](https://github.com/equinor/komodo-releases/blob/main/jenkins/spe1/spe1.ert)

```
QUEUE_SYSTEM LOCAL
QUEUE_OPTION LOCAL MAX_RUNNING 50

RUNPATH spe1_out/real_%d/iter_%d

NUM_REALIZATIONS 2

ECLBASE eclipse/SPE1CASE1


FORWARD_MODEL COPY_DIRECTORY([FROM](https://app.slack.com/client/T02JL00JU/CKC3H3W74/thread/FROM)=[CONFIG_PATH](https://app.slack.com/client/T02JL00JU/CKC3H3W74/thread/CONFIG_PATH)/opm-data/spe1, [TO](https://app.slack.com/client/T02JL00JU/CKC3H3W74/thread/TO)=[RUNPATH](https://app.slack.com/client/T02JL00JU/CKC3H3W74/thread/RUNPATH)/eclipse)

FORWARD_MODEL ECLIPSE100([VERSION](https://app.slack.com/client/T02JL00JU/CKC3H3W74/thread/VERSION)=2017.1)
```

### PR DESCRIPTION
**Pull request by andreas-el**
**Issue**
Resolves #4569 

Should this be labeled as bug-fix? Just maintenance?

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py [hunk 1]
```diff
@@ -267,3 +267,31 @@ def test_that_data_file_sets_num_cpu(eclipse_data, expected_cpus):
...
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_runpath_substitution_remain_valid():
+    """
+    This checks that runpath substitution remain intact.
+    """
+    config_text = dedent(
+        """
+        NUM_REALIZATIONS 2
+        JOBNAME my_case%d
+        RUNPATH real_%d/iter_%d
```
Comment (berland) [inline tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py:281]: Suggest to change to `realization-%d/iter-%d` to be consistent with other files (real_%d.. is legacy)

File: tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py [hunk 2]
```diff
@@ -267,3 +267,31 @@ def test_that_data_file_sets_num_cpu(eclipse_data, expected_cpus):
...
+    )
+    Path("config.ert").write_text(config_text)
+
+    res_config = ResConfig("config.ert")
+    ert = EnKFMain(res_config)
+
+    run_context = ert.create_ensemble_context("prior", [True, True], iteration=0)
+    ert.createRunPath(run_context)
+
+    for i, runcontext in enumerate(run_context):
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py:293]: The enumeration of `run_context` should be `run_arg` (or perhaps `realization`)

File: tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py [hunk 3]
```diff
@@ -267,3 +267,31 @@ def test_that_data_file_sets_num_cpu(eclipse_data, expected_cpus):
...
+        assert (
+            "real_" + str(i) + "/iter_0"
+            in Path(runcontext.runpath + "/jobs.json").read_text()
+        )
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py:297]: Could consider making this into a snapshot test?

```suggestion
snapshot.assert_match(Path(run_context[0].runpath + "/jobs.json").read_text(), "jobs.json_realization_0`)
snapshot.assert_match(Path(run_context[1].runpath + "/jobs.json").read_text(), "jobs.json_realization_1`)

**Comment by oyvindeide** [comment]
Looks really good (thanks for fixing my mistakes), just some minor comments

File: tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py (hunk 3 above)
Comment (andreas-el) [reply tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py]: This is a good idea, but at the moment a bit challenging as the file is constantly changing due to ert_pid and others.

File: tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py (hunk 1 above)
Comment (andreas-el) [reply tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py]: Added!

File: tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py [hunk 4]
```diff
@@ -267,3 +267,31 @@ def test_that_data_file_sets_num_cpu(eclipse_data, expected_cpus):
...
+
+    res_config = ResConfig("config.ert")
+    ert = EnKFMain(res_config)
+
+    run_context = ert.create_ensemble_context("prior", [True, True], iteration=0)
+    ert.createRunPath(run_context)
+
+    for i, realization in enumerate(run_context):
+        assert (
+            "realization-" + str(i) + "/iter-0"
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py:295]: Perhaps expand this even more (just to guard agains accidental false positives?

**Review by oyvindeide — APPROVED** [review]
LGTM! Just one minor comment, remember to squash into one commit before merging

File: tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py (hunk 4 above)
Comment (andreas-el) [reply tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py]: Full path it is. :+1:

**Pull request merged**